File: src/backend/Intelibill.Application/Features/Suppliers/DTOs/SupplierDto.cs
namespace Intelibill.Application.Features.Suppliers.DTOs;

using Intelibill.Domain.Enums;

public sealed record SupplierDto(
    Guid SupplierId,
    string Name,
    string? ContactPersonName,
    string? ContactPersonPhone,
    string? Address,
    string? City,
    string? State,
    string? Pin,
    bool IsSystem,
    bool IsActive,
    bool IsPreferred,
    decimal BalanceDue);

File: src/mobile/android/intelibill_mobile/lib/src/features/suppliers/data/dto/supplier_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'supplier_dto.freezed.dart';
part 'supplier_dto.g.dart';

@freezed
sealed class SupplierDto with _$SupplierDto {
  const factory SupplierDto({
    @JsonKey(name: 'supplierId') required String supplierId,
    @JsonKey(name: 'name') required String name,
    @JsonKey(name: 'contactPersonName') String? contactPersonName,
    @JsonKey(name: 'contactPersonPhone') String? contactPersonPhone,
    @JsonKey(name: 'address') String? address,
    @JsonKey(name: 'city') String? city,
    @JsonKey(name: 'state') String? state,
    @JsonKey(name: 'pin') String? pin,
    @JsonKey(name: 'isSystem') required bool isSystem,
    @JsonKey(name: 'isActive') required bool isActive,
    @JsonKey(name: 'isPreferred') required bool isPreferred,
    @JsonKey(name: 'balanceDue') @Default(0.0) double balanceDue,
  }) = _SupplierDto;

  factory SupplierDto.fromJson(Map<String, dynamic> json) =>
      _$SupplierDtoFromJson(json);
}
